File: 3차 과제/4-2/py 소스파일/4-2.py
from node4_2 import listnode
from link_list4_2 import startNode
from func4_2 import print_func
from func4_2 import search
from func4_2 import delete
from func4_2 import reverse_print

def main() :
    week = startNode()
    print("(1)리스트에 [월], [수], [일] 노드 삽입하기")
    week.insertfirst('월')
    week.insertlast('수')
    week.insertlast('일')
    print_func(week)

    print("(2) 리스트에서 [수]노드 탐색하기!");
    p=search(week, '수')
    print(f"[{p.data}]를 찾았습니다.\n")

    print("(3) 리스트의 [수]노드 뒤에 [금] 노드 삽입하기!");
    week.insertmiddle(p, '금')
    print_func(week)

    print("(4) 리스트에 [일]노드 삭제하기!");
    p=search(week, '일')
    delete(week, p)
    print_func(week)

    print("(5) 리스트의 순서를 역순으로 바꾸기!");
    reverse_print(week)
    
    week.linked_link_Free()

if __name__ == "__main__":
    main()

File: DataStructure/3차 과제/4-2/c 소스파일/4-2.c
#include <stdio.h>
#include "4-2.h"

int main() {
    startNode* week = Empty_limked_link(); // 공백 리스트 만들기(시작 리스트)
    listNode* p;
    printf("(1)리스트에 [월], [수], [일]노드 삽입하기!\n");
    insertFirst(week, "수");
    insertList(week, "일");
    insertFirst(week, "월");
    print(week);

    printf("(2) 리스트에서 [수]노드 탐색하기!\n");
    p=search(week, "수");
    if (p==NULL) printf("찾는 데이터가 없습니다.\n");
    else printf("[%s]를 찾았습니다.\n\n", p->data);

    printf("(3) 리스트에 [수]노드 뒤에 [금]노드 삽입하기!\n");
    insertMiddle(week, p, "금");
    print(week);

    printf("(4) 리스트에서 [일]노드 삭제하기!\n");
    p=search(week, "일");
    delete(week, p);
    print(week);

    printf("(5) 리스트의 순서를 역순으로 바꾸기!\n");
    reverse(week);

    return 0;
}